app/app.rb

```diff
@@ -0,0 +1,84 @@
+ENV["RACK_ENV"] ||= "development"
+
+require 'sinatra/base'
+require_relative 'data_mapper_setup'
+
+class Chitter < Sinatra::Base
+
+  enable :sessions
+  set :session_secret, 'super secret'
+
+  use Rack::MethodOverride
+  register Sinatra::Flash
+
+  helpers do
+    def current_user
+      @current_user ||= User.get(session[:user_id])
+    end
+  end
+
+  get '/' do
+    erb :'welcome'
+  end
+
+  get '/create_user' do
+    @user = User.new
+    erb :create_user
+  end
+
+  post '/create_user' do
+    @user = User.create(name: params[:name],
+                        username: params[:username],
+                        email: params[:email],
+                        password: params[:password],
+                        password_confirmation: params[:password_confirmation])
+    if @user.save
+      session[:user_id] = @user.id
+      redirect '/'
+    else
+      flash.now[:errors] = @user.errors.full_messages
+      erb :create_user
+    end
+  end
+
+  get '/log_in' do
```

Comment:
should be `/sessions/new` or similar to be RESTful; consider also your `post` and `delete` routes.